process/economics/economicsData.go

```diff
@@ -431,6 +446,17 @@ func (ed *economicsData) EpochConfirmed(epoch uint32) {
 
 	ed.flagGasPriceModifier.Toggle(epoch >= ed.gasPriceModifierEnableEpoch)
 	log.Debug("economics: gas price modifier", "enabled", ed.flagGasPriceModifier.IsSet())
+	ed.setGasPriceModifierMetric()
+}
+
+func (ed *economicsData) setGasPriceModifierMetric() {
+	if check.IfNil(ed.statusHandler) {
+		log.Debug("cannot set GasPriceModifier metric - nil status handler")
```

Comment:
How is this possible? That was one of the reason we used nil status handler. To get rid of these checks.

Reply:
Technically it is not possible if economicsData is used right. Ok, will remove the check